### PR TITLE
Added macros conan_check and conan_add_remote.

### LINKED ISSUE #92: Missing support for adding a conan-remote inside CMake
Hi,

I am currently figuring out how to do this with 
```
execute_process(
	COMMAND "conan remote add ...)
```
but it would be nice to have a specific function inside the conan.cmake as well.

### PR DESCRIPTION
**Pull request by bilke**
Example usage:

```cmake
conan_check(VERSION 1.0.0 REQUIRED)
conan_add_remote(NAME bincrafters INDEX 1
    URL https://api.bintray.com/conan/bincrafters/public-conan)
```

Closes #92.

### CODE REVIEW

File: conan.cmake [hunk 1]
```diff
@@ -426,3 +426,60 @@ macro(conan_cmake_run)
...
+macro(conan_add_remote)
+    # Adds a remote
+    # Arguments URL and NAME are required, INDEX is optional
+    # Example usage:
+    #    conan_add_remote(NAME bincrafters INDEX 1
+    #       URL https://api.bintray.com/conan/bincrafters/public-conan)
+    set(oneValueArgs URL NAME INDEX)
+    cmake_parse_arguments(CONAN "" "${oneValueArgs}" "" ${ARGN})
+
+    conan_check(REQUIRED)
```
Comment (memsharded) [inline conan.cmake]: I'd say that a check of conan for each invocation is too much. Probably it is something that a user can do, in their code. But adding a ``conan_check(REQUIRED)`` to each call seems overkill.

File: conan.cmake [hunk 2]
```diff
@@ -426,3 +426,60 @@ macro(conan_cmake_run)
...
+
+    if(DEFINED CONAN_INDEX)
+        set(CONAN_INDEX_ARG "-i ${CONAN_INDEX}")
+    endif()
+
+    execute_process(COMMAND ${CONAN_CMD} remote list
+        OUTPUT_VARIABLE CONAN_REMOTES)
+
+    if("${CONAN_REMOTES}" MATCHES ".*${CONAN_NAME}:.*")
```
Comment (memsharded) [inline conan.cmake]: Are you aware of the ``conan remote add --force`` argument? It does update if existing.

**Comment by bilke** [comment]
@memsharded Thanks for the hint on the `--force` option! I addressed both of your comments.

**Comment by memsharded** [comment]
Excellent, thank you! Will do a release now.

**Pull request merged**